include/fmt/core.h

```diff
@@ -98,6 +98,13 @@
 #  define FMT_CONSTEXPR_DECL
 #endif
 
+// C++17's char_traits::length() is constexpr.
+#if __cplusplus >= 201703L
+#  define FMT_CONSTEXPR_CHAR_TRAITS_LENGTH FMT_CONSTEXPR
+#else
+#  define FMT_CONSTEXPR_CHAR_TRAITS_LENGTH
+#endif
+
```

Comment:
Also I'm not sure if it's worth introducing this special definition to use it only once.